mp/src/game/client/c_baseanimating.cpp

```diff
@@ -3179,8 +3179,39 @@ int C_BaseAnimating::DrawModel( int flags )
 		{
 			extraFlags |= STUDIO_IGNORE_NEO_EFFECTS;
 		}
-#endif
 
+		auto pLocalPlayer = C_NEO_Player::GetLocalNEOPlayer();
+		Assert(pLocalPlayer);
+
+		const bool inMotionVision = pLocalPlayer && pLocalPlayer->IsInVision() && pLocalPlayer->GetClass() == NEO_CLASS_ASSAULT;
```

Comment:
I don't know if removing the Assert and leaving the defensive `pLocalPlayer && ...` in the boolean assignment does us any good, since now we're hiding any nullptr dereference bug where `!pLocalPlayer` from triggering the removed assert, but still suffer the overhead from having to do the `pLocalPlayer && ...` comparison for every draw call at the boolean assignment. I'd personally prefer keeping the assert, and instead doing:

```cpp
const bool inMotionVision = pLocalPlayer->IsInVision() && pLocalPlayer->GetClass() == NEO_CLASS_ASSAULT;
```

But probably entering the realm of micro optimization anyway, so I guess it doesn't matter too much in practice.

Reply:
Or whoops, writing comments to an outdated GitHub window. Disregard.